server/routes/payment.js

```diff
@@ -0,0 +1,19 @@
+const { Router } = require("express");
+const router = Router();
+
+const createPayment = (req, res, next) => {
+  try {
+    // verify value
+    // call Stripe API
+    // post to db
+    console.log(req.body);
+    return res.status(201).send({ result: {} });
```

Comment:
Random side note about another PR, but remembered it here - I don't think it needs to change since it really depends on the number of payments per user, but I have been thinking about the payment structure in the database and realized that we don't really need all the payments nested on the user, since we will rarely need to get the payments right on the user and it could get big actually, when we probably will have a lot of other stuff cached on the user object. 

I would have opted for a more SQLish structure and have the userId on the payment, and not nest it. You could probably index the userId on the payment so it's efficient to fetch.

Reply:
Take a look at this - https://docs.google.com/presentation/d/1taIi9ohiVNtwP2r7oPMMw6CMmQNt5UlofIJGHNdz5cE/edit#slide=id.g737e11900c_0_140 and the next slide

The design in NoSQL is that it is based on your query pattern - so essentially there are many ways to do a one-to-many relationship and it depends on when you need information and the size of different objects. Basically, there is no standard way of doing it, making it a bit more challenging to design, and a lot more flexible

Reply:
If we query to update the User credits attribute will be edited when each new Payment clears, is there any detriment in storing refs to the Payment on the User? No additional db calls, and we can keep a trail on the User for easy verification if that's ever needed. The User.toObject transform could drop the payments since we don't need them in most cases, and in those that we do just call populate?